### PR TITLE
chore(go-client-ci): build the server instead of downloading it from external site

### PR DESCRIPTION
**Pull request by acelyc111**
https://github.com/apache/incubator-pegasus/issues/1779

- Separate format and lint jobs
- Build Pegasus server binaries in the CI procedure instead of downloading it from any site
- Fix lint and test failures
- improve readme

### CODE REVIEW

**Review by GehaFearless — DISMISSED** [review]

File: go-client/session/radmin_rpc_types.go [hunk 1]
```diff
@@ -15,7 +15,7 @@
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
```
Comment (empiredan) [inline go-client/session/radmin_rpc_types.go:17]: Whether all of these lines should also be indented by one space ?

File: go-client/session/admin_rpc_types.go [hunk 2]
```diff
@@ -15,7 +15,7 @@
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
```
Comment (empiredan) [inline go-client/session/admin_rpc_types.go:17]: Whether all of these lines should also be indented by one space ?

File: go-client/test.sh [hunk 3]
```diff
@@ -1,46 +0,0 @@
...
-PEGASUS_PKG_URL="https://github.com/apache/incubator-pegasus/releases/download/v2.0.0/pegasus-tools-2.0.0-5d969e8-glibc2.12-release.tar.gz"
-
-# start pegasus onebox environment
-if [ ! -f $PEGASUS_PKG.tar.gz ]; then
-    wget --quiet $PEGASUS_PKG_URL
-    tar xf $PEGASUS_PKG.tar.gz
-fi
-cd $PEGASUS_PKG
-
-./run.sh start_onebox -m 2 -r 3 -w
```
Comment (empiredan) [inline go-client/test.sh:35]: Shall we keep a script allowing the developers to run their tests locally, or just add some instructions in README ?

Comment (acelyc111) [reply go-client/test.sh:35]: Added to README

File: go-client/session/radmin_rpc_types.go (hunk 1 above)
Comment (acelyc111) [reply go-client/session/radmin_rpc_types.go]: It is generated by go-client/generator/main.go, I've updated the files.

File: go-client/session/admin_rpc_types.go (hunk 2 above)
Comment (acelyc111) [reply go-client/session/admin_rpc_types.go]: Same.

**Review by empiredan — APPROVED** [review]

**Review by GehaFearless — APPROVED** [review]

**Pull request merged**